File: backend/ai_service/app/models/session.py
"""SQLModel read-side mirror of Prisma table "Session"."""

from datetime import datetime

from sqlmodel import Field, SQLModel

from app.models.timestamps import utcnow


class Session(SQLModel, table=True):
    """Group recommendation session; mirrors Prisma model Session."""

    __tablename__ = "Session"

    id: int | None = Field(default=None, primary_key=True)
    host_user_id: int = Field(foreign_key="User.id")
    group_id: int | None = Field(default=None, foreign_key="Group.id")
    time_limit: int
    # The host's chosen event time (from the pre-session modal). Drives restaurant
    # open/closed evaluation in the orchestrator; snapshotted onto Event.date at
    # close. Replaces the removed host Qa.time_slot free-text field.
    scheduled_for: datetime | None = None
    created_at: datetime = Field(default_factory=utcnow)
    closed_at: datetime | None = None
    # No avg_budget column: the averaged group budget is computed on demand from
    # members' effective budget_max by the orchestrator, never persisted.

File: frontend/src/types/session.ts
// Modeled on Prisma `Session` + `SessionMember`.
// NOTE: Prisma has NO session-status enum. Per-member readiness is
// `SessionMember.status: boolean`. `SessionPhase` below is FRONTEND-ONLY UI
// state derived from members/recommendation — it is never persisted.

export interface Session {
  id: number
  host_user_id: number
  group_id?: number | null
  time_limit: number
  // The host's chosen event time (pre-session modal). Drives the open/closed
  // hard filter server-side and is snapshotted onto Event.date at close. Null on
  // legacy sessions created before the field existed.
  scheduled_for?: string | null
  created_at: string
  closed_at?: string | null
}

export interface SessionMember {
  // The gateway's getSession/listMembers reads DON'T return session_id (it's
  // implied by the route), but they DO return the member's display_name. Model
  // the actual wire shape: session_id optional, display_name present.
  session_id?: number
  user_id: number
  display_name?: string | null
  status: boolean // true = done sharing preferences
  joined_at: string
}

// FRONTEND-ONLY: drives which screen shows. Not a DB field.
export type SessionPhase =
  | 'joining'
  | 'waiting'
  | 'chatting'
  | 'done'
  | 'picks'
  | 'detail'
  | 'complete'
